server/src/infra/ldap_handler.rs

```diff
@@ -120,15 +120,54 @@ fn get_user_attribute(user: &User, attribute: &str, dn: &str) -> Result<Option<V
     }))
 }
 
+fn expand_attribute_wildcards(
+    attributes: &[String],
+    all_attribute_keys: &[&str],
+    operational_attribute_keys: &[&str],
+) -> Vec<String> {
+    let mut attributes_out = attributes.to_owned();
+
+    if let Some(all_idx) = attributes_out.iter().position(|x| x == "*") {
+        debug!(r#"Expanding *"#);
+        // Splice in all non-operational attributes
+        attributes_out.splice(
+            all_idx..all_idx + 1,
+            all_attribute_keys
+                .into_iter()
+                .cloned()
```

Comment:
I'm not sure you need this clone. and `into_iter` can probably be just `iter` since it's readonly.

Reply:
you're probably right about the `into_iter` but i definitely chained some of these for a reason (make compiler happy)

Reply:
oh, actually, i think the `.map` call is probably what made compiler happy